### ISSUE TITLE
Overloader getting stuck outside of the field

### ISSUE THREAD

**Issue by Randy1214** [Courseplay/courseplay#2873]
Is there a way to restrict the overloader from going outside of the field border? Right now i have to overload my own combines because the overloader tractor gets stuck every 5 minutes. When it turns around it gets stuck on trees on the edge of the field or when its full and wants to go back to the waiting trailer it tries to take the shortest route and ends up in the forest, especially if the field is an irregular shape like a U. 

If there would be an option to only allow the tractor to turn around inside the field it would fix this issue.

**Comment by Randy1214** [comment]
Id like to add that the course generation for seeding and fieldwork already does this. Turns inside the field and follow the inside of the field to go to their next block so why not that option to the overload mode

**Comment by pops64** [comment]
No. The 5.3 version of modhub should also include this feature. Tho there have been tweaks to it since 5.3

**Comment by Randy1214** [comment]
For seeding, fertilizing, harvesting and any other field work they all turn inside the field and it works great the only mode i had trouble with is the overload mode.

**Comment by pops64** [comment]
Are you talking about the very first few rows for chaff havesters? Has this mode isn't designed to keep it in the field on those cases. The combine havester will keep it in the field expect in cases when the field next to it is close enough that it thinks it is turning inside the field. Which is on my list to fix

**Comment by Randy1214** [comment]
No im talking about the tractor that automatically searches for combines in the field and drive to them to unload. Often times the tractor will need to turn around to go to the next and gets stuck in trees or anything on the outside of the field. Instead of waiting for the combine to turn around, it either drives in trees or cuts in front of the combine and then both are stuck in traffic

**Comment by pops64** [comment]
I need a picture or video. I know if you have a irregular field shape CP will drive straight for the combine. I know there is a bug I am currently trying to fix that the tipper will try to unload the combine in the middle of a turn. That could cause issues but not every few minutes

**Comment by Randy1214** [comment]
Ill get you a video when i get home

**Comment by pops64** [comment]
Alright it doesn't sound normal. I am playing on Pacific Northwest ATM and I already have issues. The only bug not list here is when it's parked close to the edge it will turn towards edge instead of into the field which will occasionally cause it to hit the hedge

**Comment by Randy1214** [comment]
Thats mostly what my issue has been or when going to the unload trailor it takes the shortest route and sometimes thats threw the forest. That and sometimes when it unloaded the first combine it parks in the way of the second one

**Comment by pops64** [comment]
It shouldn't be parking in the way of another combine. It should always be parking in the middle of the last lane used by the combine it is unloading. Unless you're using a small header(5ms or less usually) The forset one is cause of irregular shape fields. Which I can fix I just don't know where to place that option in the HUD as it is CPU heavy and can cause a frame rate hang

**Comment by Randy1214** [comment]
I am using  small headers so that issue will be solved when i get money to buy better stuff. The second problem would be awesome if it was fixed. Thanks for quick replies btw

**Comment by Randy1214** [comment]
Is there a way you can try to fix my second problem and send me a copy so i can try it out and see how my franes rates are?

**Comment by pops64** [comment]
[mode2.zip](https://github.com/Courseplay/courseplay/files/2451858/mode2.zip)

Sorry for the delay replace mode2.lua with that file it should be stable but is untested

**Comment by Randy1214** [comment]
I use the overloader mode which is mode 3 because i use a overload trailer to dump into another waiting trailer which then has his own course to go unload at the silo. 

Is it hard for you to adapt that same code for mode 3?

Cause from what i understand in the manual both modes are very similar they just unload differently

**Comment by pops64** [comment]
the only difference between mode 2 and 3 is when driving the unload course. So this file will keep any tipper/overloader on the field when driving to combine. This won't work driving to the unload course atm that requires a different change which ill look into at later time

**Comment by Randy1214** [comment]
So if i replace mode2 with the file you gave me will that also have an effect on mode3 or do i have to use mode2 to see the changes you made
